### ISSUE TITLE
Files containing byte order mark (BOM) are not supported

### ISSUE THREAD

**Issue by darfink** [arcnmx/serde-ini#14]
Albeit BOMs are redundant for UTF-8, they are still commonly encountered.
I personally tried to parse an UTF-8 with a BOM, which lead to time consuming troubleshooting.
It would be great if a potential BOM is present, it is discarded instead of resulting in an error.

**Comment by arcnmx** [comment]
Yeah, I'm not entirely sure if this is the right abstraction level, or if it should be the responsibility of a `Read` wrapper to handle skipping past BOMs instead (or be extra fancy and also transparently convert non-utf8). It's not a big deal to just add a check for it though...

From what I can gather, winapi's `GetPrivateProfileString` also chokes on a UTF8 BOM, so this isn't inconsistent. It might support UTF-16 (UCS-2?) BOMs though? Or only `GetPrivateProfileStringW` might, I'm not sure...

**Comment by darfink** [comment]
Interesting! Well depending on whether it's the library's responsibility or not, you can consider this issue open or closed.
Meanwhile, I decided to use [encoding_rs_io](https://docs.rs/encoding_rs_io/0.1.7/encoding_rs_io/) to handle files with UTF8 BOMs.
